### PR TITLE
NMS-16341: license validation and formatting updates

### PR DESCRIPTION
**Pull request by RangerRick**
This PR modifies the existing source copyright checking to use a new plugin, and a new header format.

I am adopting Apache's style of header management, which has generic copyright assignment wording and points to a top-level document that goes into more detail.

There are separate commits for turning on the new license-handling code, and for actually reformatting the code.

One advantage to using this plugin is now you can run: `./compile.pl license:format` and it will automatically format any new files you create before committing. (woooo!)

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-16341

### CODE REVIEW

**Comment by RangerRick** [comment]
(also this PR gets rid of the `solaris/ directory which hasn't been maintained for eons)

File: pom.xml [hunk 1]
```diff
@@ -1058,6 +1081,86 @@
...
+            <owner>The OpenNMS Group, Inc.</owner>
+            <email>license@opennms.com</email>
+          </properties>
+          <licenseSets>
+            <licenseSet>
+              <header>src/main/resources/license-header.txt</header>
+              <!-- some of these excludes are aspirational, for when we expand this beyond the basics 😅 -->
+              <excludes>
+                <!-- short term, let's exclude everything but c/c++/groovy/java/js/ts -->
+                <exclude>**/*.css</exclude>
```
Comment (synqotik) [inline pom.xml:1099]: Should we also exclude `.tsx` files?

**Review by synqotik — APPROVED** [review]
Added a small comment, but seems good to me!

File: pom.xml (hunk 1 above)
Comment (RangerRick) [reply pom.xml:1099]: I mean... we don't have any?  :)

```
~/git/opennms-develop $ find * -name \*.tsx
~/git/opennms-develop $
```

And ultimately I'd like to add the other formats too, there just needs to be some work on excluding some things that don't need anything added.

**Pull request merged**